resources/services/cloudtrail/trails.go

```diff
@@ -225,47 +226,64 @@ func CloudtrailTrails() *schema.Table {
 func fetchCloudtrailTrails(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
 	c := meta.(*client.Client)
 	svc := c.Services().Cloudtrail
+	log := meta.(*client.Client).Logger()
 	response, err := svc.DescribeTrails(ctx, nil, func(options *cloudtrail.Options) {
 		options.Region = c.Region
 	})
+
 	if err != nil {
 		return err
 	}
 
-	processTrailsBundle := func(trails []types.Trail, region string) error {
+	getBundledTrailsWithTags := func(trails []types.Trail, region string) ([]CloudTrailWrapper, error) {
+		processed := make([]CloudTrailWrapper, len(trails))
+
 		input := cloudtrail.ListTagsInput{
 			ResourceIdList: make([]string, 0, len(trails)),
 		}
-		for _, h := range trails {
-			input.ResourceIdList = append(input.ResourceIdList, *h.TrailARN)
-		}
-		for {
-			response, err := svc.ListTags(ctx, &input, func(options *cloudtrail.Options) {
-				options.Region = region
-			})
+
+		for i, h := range trails {
+			processed[i] = CloudTrailWrapper{
+				Trail: h,
+				Tags:  make(map[string]interface{}),
+			}
+
+			// Before fetching trail tags we have to check if the trail is organization trail
+			// If the trail is organization trail and the account id is not matched with current account id
+			// We skip, and not fetch the trail tags
+			arnParts, err := arn.Parse(*h.TrailARN)
 			if err != nil {
-				return err
+				log.Warn("cloud not parse cloudtrail ARN", "arn", *h.TrailARN)
+				continue
 			}
-			if len(response.ResourceTagList) == 0 {
-				break
+			if *h.IsOrganizationTrail && c.AccountID != arnParts.AccountID {
+				log.Warn("the trail is an organization level trail, cloud not fetch tags", "arn", *h.TrailARN)
+				continue
 			}
-			for _, tr := range trails {
-				wrapper := CloudTrailWrapper{
-					Trail: tr,
-					Tags:  make(map[string]interface{}),
+
+			input.ResourceIdList = append(input.ResourceIdList, *h.TrailARN)
+		}
+
+		if len(input.ResourceIdList) > 0 {
+			for {
+				response, err := svc.ListTags(ctx, &input, func(options *cloudtrail.Options) {
+					options.Region = region
+				})
+				if err != nil {
+					return nil, err
 				}
-				for _, t := range getCloudTrailTagsByResourceID(*tr.TrailARN, response.ResourceTagList) {
-					wrapper.Tags[*t.Key] = t.Value
+				for _, tr := range processed {
+					for _, t := range getCloudTrailTagsByResourceID(*tr.TrailARN, response.ResourceTagList) {
+						tr.Tags[*t.Key] = t.Value
+					}
 				}
-				res <- wrapper
-			}
-
-			if aws.ToString(response.NextToken) == "" {
-				break
+				if aws.ToString(response.NextToken) == "" {
+					break
+				}
+				input.NextToken = response.NextToken
 			}
```

Comment:
Can we just return early if input list is empty?

Then you can do a while loop for  `response.NextToken  == ""`